### PR TITLE
Update RBAC and templates for notebook-controller

### PR DESCRIPTION
**Pull request by 4n4nd**
Make standalone notebook-controller work on Openshift
Add instructions to do so
#4 

@HumairAK @cfchase Could you folks please try these instructions out?
I tested them using crc

### CODE REVIEW

File: components/notebook-controller/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 # Image URL to use all building/pushing image targets
-IMG ?= gcr.io/kubeflow-images-public/notebook-controller
+IMG ?= quay.io/4n4nd/notebook-controller
```
Comment (cfchase) [inline components/notebook-controller/Makefile:3]: Let's choose a common quay.io org.  I know of [opendatahub](https://quay.io/organization/opendatahub).  Does AICoE have one?

Comment (cfchase) [reply components/notebook-controller/Makefile:3]: also, can't we just override this with an env var?

**Comment by cfchase** [comment]
Perhaps we should start a feature branch in our org that will later get offered up to the greater kubeflow community instead of trying to merge this into master

File: components/notebook-controller/Makefile (hunk 1 above)
Comment (4n4nd) [reply components/notebook-controller/Makefile:3]: > also, can't we just override this with an env var?

yeah I didn't really think about it

**Comment by 4n4nd** [comment]
> Perhaps we should start a feature branch in our org that will later get offered up to the greater kubeflow community instead of trying to merge this into master

Yeah let's create a new branch for this :100:

File: components/notebook-controller/Makefile [hunk 2]
```diff
@@ -61,7 +61,7 @@ generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
 
 # Build the docker image
-docker-build: test
+docker-build:
```
Comment (cfchase) [inline components/notebook-controller/Makefile:64]: if it's just the `check-license` portion that's failing, perhaps we can just omit that until we [update the licenses](https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/go-license-tools/README.md)

**Comment by cfchase** [comment]
> > Perhaps we should start a feature branch in our org that will later get offered up to the greater kubeflow community instead of trying to merge this into master
> 
> Yeah let's create a new branch for this

I don't have access. @durandom will have to do it

File: components/notebook-controller/Makefile (hunk 2 above)
Comment (4n4nd) [reply components/notebook-controller/Makefile:64]: yes it was the check-license part that’s failing

**Comment by 4n4nd** [comment]
I just created a branch called pvc

File: components/notebook-controller/oc-setup.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* Lets create a PVC that we will attach to our Notebook pod: <br>
+  `oc apply -f loadtest/jupyter_pvc.yaml -n notebook-test`
+
+* Now we need to create a Notebook CR: <br>
+  `oc apply -f loadtest/jupyter_test.yaml -n notebook-test`
+
+  If the Notebook-Controller has been set up properly, this should create a new StatefulSet in the `notebook-test` namespace.
+
+* To test if the Notebook server works, we can expose the `jupyter-test` Service that was created with the StatefulSet: <br>
+  `oc expose svc jupyter-test --path=/notebook/test-notebook/jupyter-test -n notebook-test`
```
Comment (cfchase) [inline components/notebook-controller/oc-setup.md:42]: Might want to use `notebook-test` here.  switching to `test-notebook` threw me.
`oc expose svc jupyter-test --path=/notebook/notebook-test/jupyter-test -n notebook-test`

**Comment by cfchase** [comment]
> Make standalone notebook-controller work on Openshift
> Add instructions to do so
> #4
> 
> @HumairAK @cfchase Could you folks please try these instructions out?
> I tested them using crc

worked!
Screenshot from 2020-06-18 01-15-37

**Review by cfchase — APPROVED** [review]

**Comment by cfchase** [comment]
> I just created a branch called pvc

cool, you want to change this PR's destination to that branch, I assume

**Comment by 4n4nd** [comment]
Awesome, so you were able to test it on OpenShift?

>cool, you want to change this PR's destination to that branch, I assume
I'll do that

**Comment by cfchase** [comment]
> Awesome, so you were able to test it on OpenShift?

Yeah, on CRC.  Trying to go back through and see what they're doing as far as vanilla kube and code generation and why things don't work "out of the box".

**Comment by 4n4nd** [comment]
@durandom Could you please create a project on quay/aicoe for this?
`quay.io/aicoe/notebook-controller`

**Comment by cfchase** [comment]
> > Awesome, so you were able to test it on OpenShift?
> 
> Yeah, on CRC. Trying to go back through and see what they're doing as far as vanilla kube and code generation and why things don't work "out of the box".

Just to clarify, this is great for our experimentation, but this is not something we'll be offering upstream.  This would not be appropriate for that purpose.

**Comment by durandom** [comment]
> @durandom Could you please create a project on quay/aicoe for this?
> `quay.io/aicoe/notebook-controller`

I don't have access to the aicoe org on quay, @goern ?

**Pull request closed without merging**